lib/cocoapods/target/build_settings.rb

```diff
@@ -0,0 +1,679 @@
+            paths.concat framework_search_paths_to_import
+          else
+            paths.concat vendored_framework_search_paths
+            paths.delete(target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE))
+          end
+          paths
+        end
+
+        define_build_settings_method :vendored_framework_search_paths, :memoized => true do
+          file_accessors.flat_map(&:vendored_frameworks).map { |f| File.join '${PODS_ROOT}', f.dirname.relative_path_from(target.sandbox.root) }
+        end
+
+        define_build_settings_method :framework_search_paths_to_import, :memoized => true do
+          return vendored_framework_search_paths unless target.requires_frameworks? && target.should_build?
+
+          vendored_framework_search_paths + [target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE)]
+        end
+
+        define_build_settings_method :other_swift_flags, :build_setting => true, :memoized => true do
+          return unless target.uses_swift?
+          flags = super()
+          flags << '-suppress-warnings' if target.inhibit_warnings?
+          if !target.requires_frameworks? && target.defines_module? && !test_xcconfig?
+            flags.concat %w( -import-underlying-module -Xcc -fmodule-map-file=${SRCROOT}/${MODULEMAP_FILE} )
+          end
+          flags
+        end
+
+        define_build_settings_method :swift_include_paths, :build_setting => true, :memoized => true, :sorted => true, :uniqued => true do
+          paths = dependent_targets.flat_map { |t| t.build_settings.swift_include_paths_to_import }
+          paths.concat swift_include_paths_to_import if test_xcconfig?
+          paths
+        end
+
+        define_build_settings_method :swift_include_paths_to_import, :memoized => true do
+          return [] unless target.uses_swift? && !target.requires_frameworks?
+
+          [target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE)]
+        end
+
+        define_build_settings_method :pods_target_srcroot, :build_setting => true do
+          target.pod_target_srcroot
+        end
+
+        define_build_settings_method :skip_install, :build_setting => true do
+          'YES'
+        end
+
+        def requires_objc_linker_flag?
+          test_xcconfig?
+        end
+
+        define_build_settings_method :requires_fobjc_arc?, :memoized => true do
+          target.podfile.set_arc_compatibility_flag? &&
+          file_accessors.any? { |fa| fa.spec_consumer.requires_arc? }
+        end
+
+        define_build_settings_method :product_bundle_identifier, :build_setting => true do
+          'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}'
+        end
+
+        define_build_settings_method :configuration_build_dir, :build_setting => true, :memoized => true do
+          return if test_xcconfig?
+          target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE)
+        end
+
+        define_build_settings_method :dependent_targets, :memoized => true do
```

Comment:
there is a bit of ambiguity and confusion with having `define_build_settings_method` and `:build_setting => true`.

Both of them make me think that they will end up in the final `xcconfig` where I think in reality only the latter one is the one that actually forces it to be part of the `xcconfig`?

Reply:
The reason I only asked in this line is because I knew `dependent_targets` should not be part of the xcconfig and it is not but that was the actual confusion

Reply:
also I think `dependent_targets` is also memoized by `pod_target.rb`

Reply:
`dependent_targets` is recursive, it's a different method

> Both of them make me think that they will end up in the final xcconfig where I think in reality only the latter one is the one that actually forces it to be part of the xcconfig?
 
it's the latter, I can rename the method if we come up with something better